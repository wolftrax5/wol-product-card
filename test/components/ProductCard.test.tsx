import React from 'react';
import render from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';

const { act } = render;

describe('Product Card', () => {
  test('debe de mostrar la Card Personalizado', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>
        {() => <h1>Product Card</h1>}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe incrementar el contador,', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>plus</button>
          </>
        )}
      </ProductCard>
    );
    let tree = wrapper.toJSON();

    console.log((tree as any).children[2].props);
    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('0');

    expect(tree).toMatchSnapshot();
  });
});
