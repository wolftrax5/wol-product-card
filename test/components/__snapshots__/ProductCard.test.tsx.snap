// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card debe de mostrar la Card Personalizado 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`Product Card debe incrementar el contador, 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    plus
  </button>
</div>
`;
